File: src/main/java/io/github/lncvrt/lncvrtbox/events/PrepareAnvilListener.java
package io.github.lncvrt.lncvrtbox.events;

import org.bukkit.Material;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.PrepareAnvilEvent;
import org.bukkit.inventory.ItemStack;

public class PrepareAnvilListener implements Listener {
    @EventHandler
    public void onPrepareAnvil(PrepareAnvilEvent event) {
        ItemStack left = event.getInventory().getItem(0);
        ItemStack right = event.getInventory().getItem(1);

        if (left != null && (left.getType() == Material.DIAMOND_PICKAXE || left.getType() == Material.NETHERITE_PICKAXE)) {
            event.setResult(null);
        }
        if (right != null && (right.getType() == Material.DIAMOND_PICKAXE || right.getType() == Material.NETHERITE_PICKAXE)) {
            event.setResult(null);
        }
    }
}

File: src/main/kotlin/xyz/lncvrt/galaxyboxpvp/events/PrepareAnvilListener.kt
package xyz.lncvrt.galaxyboxpvp.events

import org.bukkit.Material
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.PrepareAnvilEvent

class PrepareAnvilListener : Listener {
    @EventHandler
    fun onPrepareAnvil(event: PrepareAnvilEvent) {
        val left = event.inventory.getItem(0)
        val right = event.inventory.getItem(1)

        if (left != null && (left.type == Material.DIAMOND_PICKAXE || left.type == Material.NETHERITE_PICKAXE)) {
            event.result = null
        }
        if (right != null && (right.type == Material.DIAMOND_PICKAXE || right.type == Material.NETHERITE_PICKAXE)) {
            event.result = null
        }
    }
}